### PR TITLE
feat: add README.md

### PR DESCRIPTION
**Pull request by LeNooby09**

### CODE REVIEW

**Comment by bytexenon** [comment]
Malware 3:

Pull request rejected

File: README.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+- Modular design
+- Written in TypeScript for type safety
+- Fast and efficient parsing and compiling
+
+## Installation
+
+To install Lua.js, use npm:
+
+```bash
+npm install lua-js
```
Comment (bytexenon) [inline README.md:22]: It's actually not published on NPM yet.
Plus, IIRC, the name "lua-js" is already taken in the NPM registry.
So I think it'd be better to install it from source, instead (for now).

So, something like that:
```
npm install git+https://github.com/bytexenon/Lua.js
```
would work

File: README.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+```typescript
+import { Lexer, Parser, Compiler } from 'lua-js';
+
+const luaCode = `
+print("Hello, Lua!")
+`;
+
+const lexer = new Lexer(luaCode);
+const parser = new Parser(lexer);
```
Comment (bytexenon) [inline README.md:37]: You're passing Lexer object to parser, instead you should pass tokens generated from the Lexer's `lex` method -- `lexer.lex()`

**Review by bytexenon — CHANGES_REQUESTED** [review]
CHANGE IT NOW :bangbang:

File: README.md (hunk 2 above)
Comment (LeNooby09) [reply README.md:37]: i dont know what you want bro :sob:

File: README.md (hunk 1 above)
Comment (LeNooby09) [reply README.md]: ok

File: README.md (hunk 2 above)
Comment (bytexenon) [reply README.md:37]: > i dont know what you want bro 😭

READ. IT. AGAIN.

HOW CAN YOU POSSIBLY NOT UNDERSTAND THIS?! LEXER IS A COMPONENT THAT TAKES YOUR CODE AND BREAKS IT DOWN INTO TOKENS. IT'S NOT JUST SOME RANDOM OBJECT YOU CAN PASS AROUND LIKE A TOY! IT CONVERTS CODE INTO ABSTRACT SYNTAX TREES, FOR GOODNESS' SAKE!

WHY IN THE WORLD ARE YOU PASSING AN INSTANCE OF **LEXER** TO **PARSER**?! DO YOU EVEN REALIZE HOW ABSURD THAT IS?! IT'S LIKE TRYING TO USE A HAMMER TO FIX A COMPUTER! WHAT DO YOU THINK IS GOING TO HAPPEN? DO YOU THINK THE PARSER IS JUST GOING TO FIGURE IT OUT ON ITS OWN? IT DOESN'T WORK THAT WAY! GET IT TOGETHER! THIS ISN'T ROCKET SCIENCE! IT'S FUNDAMENTAL! IF YOU CAN'T EVEN GET THIS RIGHT, HOW DO YOU EXPECT TO HANDLE ANYTHING MORE COMPLEX? THIS IS FRUSTRATING BEYOND BELIEF! GET A GRIP AND FIX YOUR CODE!

ARE YOU MAD?!

Comment (bytexenon) [reply README.md:37]:
```ts

const lexer = new Lexer(luaCode);
const tokens = lexer.lex();
const parser = new Parser(tokens);
```

Comment (LeNooby09) [reply README.md:37]: bet

**Comment by bytexenon** [comment]
LOOKS GOOD TO ME

**Comment by LeNooby09** [comment]
LFGG

**Comment by bytexenon** [comment]
> LFGG

DO NOT CLOSE

**Pull request merged**